pkg/streamer/reader.go

```diff
@@ -215,14 +175,17 @@ func (r *BinlogReader) GetFilePosByGTID(ctx context.Context, filePath string, gs
 		default:
 		}
 
-		e, err := s.GetEvent(ctx)
+		ctx2, cancel := context.WithTimeout(ctx, time.Second)
+		e, err := fileReader.GetEvent(ctx2)
+		cancel()
 		if err != nil {
+			// reach end of file
+			if errors.Cause(err) == context.DeadlineExceeded {
```

Comment:
Is there any problem if the parent `ctx` deadline exceeded? How about checking `lastPos` with the file size? (may be still there are problems if the file is being written?

Reply:
use context.background() as parent ctx of ctx2?

Reply:
if use `context.background()` as the parent, can the user to cancel the process?